connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -106,6 +116,12 @@ public void onStartup(String connector) {
     }
 
     @Override
+    public void onPause(String connector) {
```

Comment:
Yeah, I realized the difference once I was re-reading the `WorkerConnector` vs `WorkerSourceTask`/`WorkerSinkTask` implementations. But I agree, we should either match fully or remove pause/resume entirely. Both seem feasible; `WorkerTask` tracks the previous state and handles start vs resume differently, and the same could be done for `WorkerConnector`, I think just by differentiating in `start()` between the `INIT` and other cases.

Re: connectors being restartable, good question. `initialize()` is unlikely to be an issue (and we could document that you should be able to restart). The case that I could imagine not working properly is if the connector initializes anything in the constructor that ends up in a bad state after `stop()`. I don't think it's unreasonable to expect this to work, though it should probably be documented.